File: simpliMvvm/src/main/java/com/trayis/simplimvvm/ui/SimpliBase.java
package com.trayis.simplimvvm.ui;

import androidx.annotation.LayoutRes;

public interface SimpliBase {

    @LayoutRes
    int getLayoutResourceId();

    void setViewModelsToView();

    int getViewModelProviderFlag();

    @interface ViewModelProviderFlag {
        int SELF = 0;
        int PARENT = 1;
        int PARENT_FRAGMENT = 3;
    }

}

File: simpliKMvvm/src/main/java/com/trayis/simplikmvvm/ui/SimpliBase.kt
package com.trayis.simplikmvvm.ui

import androidx.annotation.LayoutRes

interface SimpliBase {

    fun bindViewModels()

    @get:LayoutRes

    val layoutResourceId: Int

    fun setViewModelProviderFlag(flag: Int)

    fun getViewModelProviderFlag(): Int

    companion object {
        val SELF = 0
        val PARENT = 1
        val PARENT_FRAGMENT = 3
    }

}